ycmd/completers/language_server/language_server_completer.py

```diff
@@ -700,6 +700,42 @@ def _GoToDeclaration( self, request_data ):
       )
 
 
+  def _GoToReferences( self, request_data ):
+    request_id = self.GetServer().NextRequestId()
+    response = self.GetServer().GetResponse( request_id,
+                                             lsapi.References( request_id,
+                                                               request_data ) )
+
+    if isinstance( response[ 'result' ], list ):
+      if len( response[ 'result' ] ) > 1:
```

Comment:
I think this is exactly the same as `Definition` above, so we should extract this into a method. Something like:

```python
return LocationListToGoTo(...)
```

Reply:
It is the same, I wasn't paying much thought to the code duplication.

But since the else part is superfluous I guess you were saying to just move the `if` part into its own function. Right?

Reply:
I think the `if` part is all that's needed and it should be its own function (shared with the other method above)

So:

* Definition

```
if it's a list
   HandleListGoTo
else
  handle non-list single-entry goto
fi
```

* References

```
HandleListGoTo
```

Reply:
Okay, should be done now.